integration/meilisearch_test.go

```diff
@@ -0,0 +1,425 @@
+//go:build integration
+
+package integration_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/avast/retry-go/v4"
+	"github.com/meilisearch/meilisearch-go"
+	"github.com/metal-stack/backup-restore-sidecar/pkg/constants"
+	"github.com/metal-stack/metal-lib/pkg/pointer"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	_ "github.com/lib/pq"
+)
+
+const (
+	meilisearchIndex    = "backup-restore-sidecar"
+	meilisearchPassword = "test123!"
+)
+
+var (
+	meilisearchContainerImage = "getmeili/meilisearch:v1.3.0"
+)
+
+func Test_Meilisearch_Restore(t *testing.T) {
+	restoreFlow(t, &flowSpec{
+		databaseType:     "meilisearch",
+		sts:              meilisearchSts,
+		backingResources: meilisearchBackingResources,
+		addTestData:      addMeilisearchTestData,
+		verifyTestData:   verifyMeilisearchTestData,
+	})
+}
+
+func Test_Meilisearch_Upgrade(t *testing.T) {
+	upgradeFlow(t, &flowSpec{
+		databaseType: "meilisearch",
+		databaseImages: []string{
+			"getmeili/meilisearch:v1.1.0",
+			// "getmeili/meilisearch:v1.2.0", commented to test if two versions upgrade also work
+			"getmeili/meilisearch:v1.3.0",
+			"getmeili/meilisearch:v1.3.1",
```

Comment:
As tests are taking so long already, I think just testing patch version upgrade once should be enough.

Reply:
done